lib/rake/cpu_counter.rb

```diff
@@ -32,7 +32,7 @@ def count
     require 'rbconfig'
 
     def count
-      if defined?(Java::Java)
+      if defined?(JRUBY_VERSION) && (Java::Java rescue nil)
```

Comment:
How about to use `RUBY_PLATFORM == 'java'` instead of current imprementation?

Reply:
Sure, let me send another diff.